drivers/modem/gsm_ppp.c

```diff
@@ -357,6 +357,12 @@ void __weak gsm_ppp_application_setup(struct modem_context *context,
 static void gsm_finalize_connection(struct gsm_modem *gsm)
 {
 	int ret;
+	static int attach_retry;
+
+	/* If attach check failed, we should not redo every setup step */
```

Comment:
We should test that for both Arkessa and Tele2 sims on 7600E to make sure it works fine.
Let me know if you need to verify it later.